common/achievement/index.ts

```diff
@@ -62,10 +54,55 @@ async function trackEvent(event: ActionEvent, context: NotificationContext) {
                 value: value,
                 action: event.actionId,
                 userId: event.user.userID,
-                relation: getRelationByContext(context),
+                // TODO - get relation OR get relationId from context
+                relation: event.context.relationId ?? '',
             },
         });
     }
 
     return true;
 }
+
+async function checkUserAchievement<ID extends ActionID>(userAchievement: UserAchievementTemplate, userId: string, context: SpecificNotificationContext<ID>) {
+    if (userAchievement) {
+        const isConditionMet = await isAchievementConditionMet(userAchievement, context);
+        if (isConditionMet) {
+            let awardedAchievement: UserAchievementTemplate;
+            if (userAchievement.achievedAt === null) {
+                awardedAchievement = await awardUser(userAchievement.id, userId);
+            }
```

Comment:
this also changed since the merge. The achievement to check is just filled if a userAchievement.achievedAt equals null or the userAchievement has a value for recordValue.